### PR TITLE
[INJIMOB-2452] : Well known discovery with attribute name change

### PR DESCRIPTION
**Pull request by balachandarg-tw**
## Description

- WellKnown Discovery from Inji Mobile side
- Update credential_issuer to issuer_id

## Issue ticket number and link

> Example : [INJIMOB-2452](https://mosip.atlassian.net/browse/INJIMOB-2452)

[INJIMOB-2452]: https://mosip.atlassian.net/browse/INJIMOB-2452?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shared/openId4VCI/Utils.ts [hunk 1]
```diff
@@ -164,8 +164,10 @@ export const getCredentialIssuersWellKnownConfig = async (
 ) => {
   let fields: string[] = defaultFields;
   let matchingWellknownDetails: any;
+  const issuerConfig = await CACHED_API.fetchIssuerConfig(issuer!);
```
Comment (swatigoel) [inline shared/openId4VCI/Utils.ts:167]: same as above, why do we need to fetch issuer config

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: machines/Issuers/IssuersService.ts [hunk 2]
```diff
@@ -40,9 +40,13 @@ export const IssuersService = () => {
     },
     checkInternet: async () => await NetInfo.fetch(),
     downloadIssuerWellknown: async (context: any) => {
-      const wellknownResponse = await CACHED_API.fetchIssuerWellknownConfig(
+      const issuerConfig = await CACHED_API.fetchIssuerConfig(
```
[the remark this answers is not in the document]
Comment (balachandarg-tw) [reply machines/Issuers/IssuersService.ts]: Updated to use it from context.

File: machines/VerifiableCredential/VCItemMachine/VCItemServices.ts [hunk 3]
```diff
@@ -114,8 +114,12 @@ export const VCItemServices = model => {
       return response;
     },
     fetchIssuerWellknown: async context => {
-      const wellknownResponse = await CACHED_API.fetchIssuerWellknownConfig(
+      const issuerConfig = await CACHED_API.fetchIssuerConfig(
```
[the remark this answers is not in the document]
Comment (balachandarg-tw) [reply machines/VerifiableCredential/VCItemMachine/VCItemServices.ts]: Added issuerHost in VcMetadata for wellknown discoery.

File: shared/openId4VCI/Utils.ts (hunk 1 above)
Comment (balachandarg-tw) [reply shared/openId4VCI/Utils.ts]: Added issuerHost in VcMetadata for wellknown discovery.

File: shared/openId4VCI/Utils.ts [hunk 4]
```diff
@@ -161,11 +161,13 @@ export const getCredentialIssuersWellKnownConfig = async (
   defaultFields: string[],
   credentialConfigurationId: string,
   format: string,
+  vcMetadata: any,
```
Comment (swatigoel) [inline shared/openId4VCI/Utils.ts:164]: do we need entire vcMetadata here?
if it's needed. can't we specify type as VCMetadata instead of any?

**Review by swatigoel — CHANGES_REQUESTED** [review]

**Review by swatigoel — APPROVED** [review]

**Review by swatigoel — APPROVED** [review]

**Review by swatigoel — APPROVED** [review]

**Pull request merged**